packages/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx

```diff
@@ -551,43 +587,70 @@ export class HorizontalBarChartWithAxisBase extends React.Component<
       if (barHeight < 1) {
         return <React.Fragment key={point.x}> </React.Fragment>;
       }
-      let color: string;
+      let startColor: string;
       if (useSingleColor) {
         //if useSingle color , then check if user has given a palette or not
         // and pick the first color from that or else from our paltette.
-        color = this.props.colors ? this._createColors()(1) : getNextColor(1, 0, this.props.theme?.isInverted);
+        startColor = this.props.colors ? this._createColors()(1) : getNextColor(1, 0, this.props.theme?.isInverted);
       } else {
-        color = this.props.colors
+        startColor = this.props.colors
           ? this._createColors()(point.x)
           : getNextColor(index, 0, this.props.theme?.isInverted);
       }
 
+      startColor = point.color && !useSingleColor ? point.color : startColor;
+      let endColor = startColor;
+
+      if (this.props.enableGradient) {
+        const pointIndex = this._points.findIndex(item => item === point);
+        startColor = point.gradient?.[0] || getNextGradient(pointIndex, 0, this.props.theme?.isInverted)[0];
+        endColor = point.gradient?.[1] || getNextGradient(pointIndex, 0, this.props.theme?.isInverted)[1];
+        if (useSingleColor) {
+          startColor = getNextGradient(0, 0, this.props.theme?.isInverted)[0];
+          endColor = getNextGradient(0, 0, this.props.theme?.isInverted)[1];
+        }
+        this._points[pointIndex].color = startColor;
+      }
+
+      const gradientId = getId('HBCWA_Gradient') + `_${index}_${point.x}`;
+
       return (
-        <rect
-          transform={`translate(0,${0.5 * (yBarScale.bandwidth() - this._barHeight)})`}
-          key={point.x}
-          x={this._isRtl ? xBarScale(point.x) : this.margins.left!}
-          y={yBarScale(point.y)}
-          width={
-            this._isRtl
-              ? containerWidth - this.margins.right! - Math.max(xBarScale(point.x), 0)
-              : Math.max(xBarScale(point.x), 0) - this.margins.left!
-          }
-          height={this._barHeight}
-          aria-labelledby={`toolTip${this._calloutId}`}
-          aria-label={this._getAriaLabel(point)}
-          role="img"
-          ref={(e: SVGRectElement) => {
-            this._refCallback(e, point.legend!);
-          }}
-          onClick={point.onClick}
-          onMouseOver={this._onBarHover.bind(this, point, color)}
-          onMouseLeave={this._onBarLeave}
-          onBlur={this._onBarLeave}
-          data-is-focusable={true}
-          onFocus={this._onBarFocus.bind(this, point, index, color)}
-          fill={point.color && !useSingleColor ? point.color : color}
-        />
+        <React.Fragment key={point.x}>
```

Comment:
Dont use x value as the key. It could duplicate for multiple bars

Reply:
Updated to `${index}_${point.x}`